cusub_perception/detection_tree/include/detection_tree/detection_tree_node.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef DETECTION_TREE_NODE_SRC_DETECTION_TREE_NODE_H_
+#define DETECTION_TREE_NODE_SRC_DETECTION_TREE_NODE_H_
+
+// Logging Formats
+#include <string>
+#define DETECTION_TREE_NAME             std::string("[\033[92mDetection Tree\033[0m] ")
+#define DETECTION_TREE_WARN_START       std::string("\033[93m[WARN] ")
+#define DETECTION_TREE_COLOR_START      std::string("\033[95m")
+#define DETECTION_TREE_END              std::string("\033[0m")
+
+# define DOBJECT_PROBABILITY_ZERO       0.0
+# define DOBJECT_NOT_FOUND              -1
+
+#include <pluginlib/class_list_macros.h>
+#include "detection_tree/detection_tree.hpp"
+#include <nodelet/nodelet.h>
+#include <ros/ros.h>
+#include <ros/timer.h>
+#include <vector>
+#include <map>
+#include <string>
+#include <cmath>
+#include <darknet_ros_msgs/BoundingBoxes.h>
+#include <darknet_ros_msgs/BoundingBox.h>
+#include <std_msgs/Header.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/PoseArray.h>
+#include <geometry_msgs/Quaternion.h>
+#include <tf/transform_listener.h>
+#include <tf/transform_datatypes.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <opencv2/calib3d.hpp>
+#include <detection_tree/Dvector.h>
+
+// Camel case classes + functions, underscore variables + namespaces
+
+namespace det_tree_ns
+{
+    class DetectionTree : public nodelet::Nodelet
```

Comment:
It seems like the tree holds on to all dvecs. Is there a reason, or would it make sense to do a "moving average" way of throwing out old ones?

Reply:
There's no reason atm beyond holding 5 or so for the averaging in void DetectionTree::averageBearing() used to average the past dvectors of a dobject in order to do association for when a new dvector comes in. The dvectors aren't a whole lot of memory but over a 30min run coming in at 60 Hz it could fill things up

Reply:
Yeah seems like a circular buffer of them would be ideal

Reply:
Depends on how much history you want to keep to run Structure from Motion later potentially

Reply:
yeah I'm sure we could do something smart in the management, especially after we've position solved for a dobject through structure from motion or some other cv technique